packages/fxa-content-server/app/scripts/views/settings/delete_account.js

```diff
@@ -121,19 +121,14 @@ var View = FormView.extend({
   },
 
   _getNumberOfProducts() {
-    let numberOfProducts = this._uniqueBrowserNames.length;
+    let numberOfProducts =
+      this._uniqueBrowserNames.length + this._activeSubscriptions.length;
     // eslint-disable-next-line no-unused-vars
     for (const client of this._attachedClients.toJSON()) {
       if (client.isOAuthApp === true) {
         numberOfProducts++;
       }
     }
-    // eslint-disable-next-line no-unused-vars
-    for (const sub of this._activeSubscriptions) {
-      if (sub.plan_id && sub.status === 'active') {
-        numberOfProducts++;
-      }
-    }
```

Comment:
We call `_getNumberOfProducts` after we filter subscriptions in `_fetchActiveSubscriptions` and set `_activeSubscriptions` so we should be able to set the number on the `.length` of the array instead of filtering through again.